astropy/io/fits/hdu/image.py

```diff
@@ -1008,7 +1008,7 @@ def __init__(self, data=None, header=None, do_not_scale_image_data=False,
     def match_header(cls, header):
         card = header.cards[0]
         return (card.keyword == 'SIMPLE' and
-                ('GROUPS' not in header or not header['GROUPS']) and
+                ('GROUPS' not in header or header['GROUPS'] is not True) and
```

Comment:
@bsipocz - since we're adding back in code that circumvents the standard, I think we may as well add it back *exactly* as it was before, i.e., `header['GROUPS'] != True` -- this will behave differently if the header returns, e.g., a `np.bool`. And then probably best to add a comment linking to this PR, as well as a `# noqa` or whatever it is, so we don't revert it in the future...